File: Python/ReverseLinkedList.py
class ListNode:
    def __init__(self,val=0,next=None):
        self.val = val
        self.next = next

def reverseLinkedList(head):
    tempNode = ListNode()
    prevNode = None

    while head:
        tempNode = head.next
        head.next = prevNode
        prevNode = head
        head = tempNode

    return prevNode

File: Javascript/ReverseLinkedList.js
function ListNode(val,next){
    this.val = (val === undefined ? 0:val)
    this.next = (next === undefined ? 0:next)
}


var reverseLinkedList = new function(head){
    let tempNode = new ListNode()
    let prevNode = null

    while(head){
        tempNode = head.next
        head.next = prevNode
        prevNode = head
        head = tempNode
    }
    return prevNode
}